pkg/service/one2onerestore/worker_tables.go

```diff
@@ -50,6 +42,36 @@ func (w *worker) restoreTables(ctx context.Context, workload []hostWorkload, key
 	}, logError)
 }
 
+func (w *worker) downloadTablesByHost(ctx context.Context, hostTask hostWorkload, keyspaces []string, refreshQueue chan<- refreshNodeInput) error {
+	manifestInfo, host := hostTask.manifestInfo, hostTask.host
+	const (
+		repeatInterval  = 10 * time.Second
+		pollIntervalSec = 10
+	)
+	defer close(refreshQueue)
+	return hostTask.manifestContent.ForEachIndexIterWithError(keyspaces, func(table backupspec.FilesMeta) error {
+		w.logger.Info(ctx, "Restoring data", "ks", table.Keyspace, "table", table.Table, "size", table.Size)
+
+		jobID, err := w.createDownloadJob(ctx, table, manifestInfo, host)
+		if err != nil {
+			return errors.Wrapf(err, "create download job: %s.%s", table.Keyspace, table.Table)
+		}
+		pr := w.downloadProgress(ctx, hostTask.host.Addr, table)
+
+		if err := w.waitJob(ctx, jobID, manifestInfo, host, pr, pollIntervalSec); err != nil {
```

Comment:
`waitJob` implementation is not a part of this PR, but it's always safer to add default case in so that we don't accidentally get stuck in an endless loop:
```go
		switch scyllaclient.RcloneJobStatus(job.Status) {
		case scyllaclient.JobError:
			return errors.Errorf("job error (%d): %s: host %s", jobID, job.Error, h.Addr)
		case scyllaclient.JobSuccess:
			w.logger.Info(ctx, "Job done",
				"job_id", jobID,
				"host", h,
				"took", time.Time(job.CompletedAt).Sub(time.Time(job.StartedAt)),
			)
			return nil
		case scyllaclient.JobRunning:
			continue
		case scyllaclient.JobNotFound:
			return errors.New("job not found")
		}
```